homeassistant/components/linkplay/__init__.py

```diff
@@ -0,0 +1,49 @@
+"""Support for LinkPlay devices."""
+
+from dataclasses import dataclass
+import logging
+
+from linkplay.bridge import LinkPlayBridge
+from linkplay.discovery import linkplay_factory_bridge
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_HOST
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import PLATFORMS
+
+
+@dataclass
+class LinkPlayData:
+    """Data for LinkPlay."""
+
+    bridge: LinkPlayBridge
+
+
+type LinkPlayConfigEntry = ConfigEntry[LinkPlayData]
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: LinkPlayConfigEntry) -> bool:
+    """Async setup hass config entry. Called when an entry has been setup."""
+
+    session = async_get_clientsession(hass)
+    if (
+        bridge := await linkplay_factory_bridge(entry.data[CONF_HOST], session)
+    ) is None:
+        _LOGGER.error(
+            "Failed to connect to LinkPlay device at %s", entry.data[CONF_HOST]
+        )
+        return False
```

Comment:
If it's failed to connect, let's retry as the device might be temporarily down:

```suggestion
        raise ConfigEntryNotReady
```